apps/condo/domains/common/hooks/useExecuteWithLock.tsx

```diff
@@ -20,5 +20,14 @@ export function useExecuteWithLock (
         })
     }, [lockName])
 
+    // Release lock when tab closes
+    useEffect(() => {
+        return () => {
+            if (releaseLockRef.current) {
+                return releaseLockRef.current()
+            }
+        }
+    }, [])
+
```

Comment:
If I'm not mistaken, `useEffect` won't work when you close the browser tab

https://stackoverflow.com/questions/61309899/useeffect-cleaning-function-is-not-called-when-tab-is-closed

Reply:
It's run when component unmounts. When tab closes, then component unmounts. Looks like it won't work sometimes. Ok, i will add handle `beforeunload`